docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1456,41 +1456,132 @@ types (32-bit and 64-bit words). The P4 language does not put any limit on the
 size of integer values, whether unsigned (`bit<W>`) or signed (`int<W>`), and it
 is up to the P4 programmer to choose the appropriate sizes. Because of this
 flexibility, P4Runtime represents P4 integer values as binary strings, using the
-bytes Protobuf type. The correct bitwidth - as per the P4 program - of each
-integer variable exposed through P4Runtime is specified in the P4Info message
-and it is up to the server to ensure that the binary string provided by the
-client has the correct length. The server must return an `INVALID_ARGUMENT`
-error code otherwise. For each integer value, the expected length in bytes is
-obtained by rounding-up the bitwidth to the nearest byte:
-`p4runtime_binary_string_length = (p4_bitwidth + 7) / 8`.
-
-For all binary strings, P4Runtime uses big-endian (&ie; network) byte-order. If
-the P4 integer type is not byte-aligned (`p4runtime_binary_string_length !=
-p4_bitwidth * 8`), P4Runtime expects the client to pad the value with unset bits
-before the most-significant bit of the value.
+`bytes` Protobuf type. The correct bitwidth - as per the P4 program - of each
+integer variable exposed through P4Runtime is specified in the P4Info message.
 
+The canonical binary string representation uses the shortest string that
+fits the encoded integer value.  This representation achieves three goals:
```

Comment:
extra space

Reply:
@teverman so I just found out that double spacing was a thing, sorry about that
I don't know if we should try to keep thing uniform at this level, it doesn't matter in the great scheme of things :)

Reply:
I'm definitely old school here out of habit :-)  Since the document is right-justified, I can't tell if Madoko eats the extra space upon conversion to PDF, but there are a few places where it looks like it does.